test/framework/src/main/java/org/elasticsearch/test/disruption/DisruptableMockTransport.java

```diff
@@ -198,8 +205,12 @@ public void run() {
                                 break;
 
                             case BLACK_HOLE:
+                                logger.trace("blackholed during response to {}: channel is {}", requestDescription, connectionStatus);
+                                onBlackholedDuringSend(requestId, action, destinationTransport);
+                                break;
                             case DISCONNECTED:
-                                logger.trace("dropping response to {}: channel is {}", requestDescription, connectionStatus);
+                                logger.trace("disconnected during response to {}: channel is {}", requestDescription, connectionStatus);
+                                onDisconnectedDuringSend(requestId, action, destinationTransport);
```

Comment:
Hmm I think I'd prefer a long delay on the response here using `onBlackholedDuringSend` too. We're using `DISCONNECTED` to indicate that the connection actively rejects the message, e.g. sends a RST, but if it rejects the response then the original requester is none the wiser and may wait for a long time before discovering the disconnect.

In practice it's almost never going to be that bad but I'd rather err on the pathological side if possible.

Reply:
++ adjusted accordingly in b181920 for both spots